File: Data Structures & Algorithms/baseball-game/submission-0.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        stack = []
        for c in operations:
            print(stack)
            if c == '+':
                result = stack[-1] + stack[-2]
                stack.append(result)
            elif c == 'D':
                result = stack[-1]  * 2
                stack.append(result)
            elif c == 'C':
                stack.pop(-1)
            else:
                stack.append(int(c))

        result = 0
        for num in stack:
            result += num
        
        return result

File: Data Structures & Algorithms/maximum-twin-sum-of-a-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int pairSum(ListNode head) {
        ListNode left = head;
        ListNode right = head;
        Stack<ListNode> stack = new Stack<>();
        
        while(right.next != null) {
            stack.add(right);
            right = right.next;
        }

        int i = 0;
        int max = 0;
        while (!stack.isEmpty()) {
            int sum = left.val + right.val;

            if (sum > max) {
                max = sum; 
            }

            left = left.next;
            right = stack.pop();
        }

        return max;
    }
}